Assets/Scripts/UI/SaveDeckButtonHandler.cs

```diff
@@ -0,0 +1,106 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.UI;
+using Contracts.PepemonCardDeck.abi.ContractDefinition;
+using System.Threading.Tasks;
+using Sirenix.OdinInspector;
+using Sirenix.Utilities;
+using Org.BouncyCastle.Utilities.Collections;
+
+public class SaveDeckButtonHandler : MonoBehaviour
+{
+    [ReadOnly] private HashSet<ulong> newSupportCards = new HashSet<ulong>();
+    [ReadOnly] private HashSet<ulong> oldSupportCards = new HashSet<ulong>();
+    [ReadOnly] private ulong currentDeckId;
+    [ReadOnly] private ulong newBattleCard;
+    [ReadOnly] public ulong oldBattleCard { get; private set; }
+
+    void Start()
+    {
+        GetComponent<Button>().onClick.AddListener(SetDeckCards);
+    }
+
+    /// <summary>
+    /// Sets the initial state used to add or remove cards on a deck
+    /// </summary>
+    /// <param name="deckId">deck being edited</param>
+    /// <param name="previousBattleCard">original battlecard of the deck</param>
+    /// <param name="previousSupportCards">original support cards of the deck</param>
+    public void PrepareForTransaction(ulong deckId, ulong previousBattleCard, List<ulong> previousSupportCards)
+    {
+        currentDeckId = deckId;
+        // initially everything is equal, and as the player selects/deselects cards it differs
+        newBattleCard = oldBattleCard = previousBattleCard;
+        oldSupportCards = new HashSet<ulong>(previousSupportCards);
+        newSupportCards = new HashSet<ulong>(previousSupportCards);
+    }
+
+    /// <summary>
+    /// Sends the transaction to set the deck cards
+    /// </summary>
+    async void SetDeckCards()
+    {
+#if UNITY_EDITOR
+        await PepemonCardDeck.SetApprovalState(true);
+#endif
+        // if a new card is not in the old list, add
+        var supportCardsToBeAdded = newSupportCards.Where(cardId => !oldSupportCards.Contains(cardId));
+        if (supportCardsToBeAdded.Count() > 0)
+        {
+            Debug.Log($"Adding {supportCardsToBeAdded.Count()} support cards to deck {currentDeckId}");
+            await PepemonCardDeck.AddSupportCards(
+                currentDeckId, 
+                supportCardsToBeAdded.Select(cardId => new SupportCardRequest { SupportCardId = cardId, Amount = 1 }).ToArray());
+        }
+
+        // if an old card is not in the new list, remove
+        var supportCardsToBeRemoved = oldSupportCards.Where(i => !newSupportCards.Contains(i));
+        if (supportCardsToBeRemoved.Count() > 0)
```

Comment:
It's possible to have duplicates of the same card in your deck, so you might need to use a dictionary instead here.
I'd probably go with something like this, where you have a count for each tokenId:
```cs
Dictionary<ulong, int> oldDeckCards;
Dictionary<ulong, int> newDeckCards;
Dictionary<ulong, int> cardsToRemove;
Dictionary<ulong, int> cardsToAdd;

...

// Get the cards to add and remove by comparing oldDeckCards and newDeckCards
for(int i = 1; i < MAX_CARD_TOKEN_ID; i++) {
    int cardCountDelta = newDeckCards[i] - oldDeckCards[i];
    if (cardCountDelta > 0) {
        cardsToAdd[i] = cardCountDelta;
    } else if (cardCountDelta < 0) {
        cardsToRemove[i] = -cardCountDelta;
    }
}

```